File: iotcloud-restapi-php/Core/src/IoTCloud/Request/GetDeviceShadowRequest.php
<?php

namespace TXIoTCloud\Request;

require_once "BaseRequest.php";

class GetDeviceShadowRequest extends CreateDeviceRequest
{

    /**
     * CreateProductRequest constructor.
     * @param $timestamp     当前unix时间戳
     * @param $nonce         随机正整数，与timestamp联合起来，用于防止重放攻击
     * @param $deviceName    设备名称
     * @param $productID     产品ID
     */
    public function __construct($timestamp, $nonce, $productID, $deviceName)
    {
        parent::__construct($timestamp, $nonce, $deviceName, $productID);
    }

}

File: iotcloud-restapi-java/Core/src/com/qcloud/restapi/iotcore/request/GetDeviceShadowRequest.java
package com.qcloud.restapi.iotcore.request;

public class GetDeviceShadowRequest extends Request {

    /**
     * 构造GetDeviceShadowRequest对象
     *
     * @param productID 从官网获得的productID
     * @param deviceName 设备名称
     */
    public GetDeviceShadowRequest(String productID, String deviceName) {
        put("Action", "GetDeviceShadow");
        put("productID", productID);
        put("deviceName", deviceName);
    }
}
